classes/ElggSchedulingPollSlot.php

```diff
@@ -25,7 +25,7 @@ public function vote(ElggUser $user) {
 			return false;
 		}
 
-		return $this->annotate('scheduling_poll_answer', true, $this->access_id, $user->guid);
+		return $this->annotate('scheduling_poll_answer', true, ACCESS_PUBLIC, $user->guid);
```

Comment:
Better would be to update the access_id of the answers when the poll access_id changes.

Even if all answers would be public from now on, there would have to be an upgrade script that updates also all the already existing answers.

Reply:
It seems that elgg v3.0.0 annotation strategy is to make their access public rather than following the entity access_id. see new like implementation in: https://github.com/Elgg/Elgg/commit/ab7336b20de0f82dca0fbb8ff8ee4dbb8cb93afe

About the upgrade script, I agree with you, we need one.